addons/website/controllers/main.py

```diff
@@ -703,32 +703,37 @@ def google_maps_api_key(self):
     # Themes
     # ------------------------------------------------------
 
+    # TODO Remove this function in Master because it only stays here for
+    # compatibility.
     def _get_customize_views(self, xml_ids):
-        View = request.env["ir.ui.view"].with_context(active_test=False)
-        if not xml_ids:
-            return View
-        domain = [("key", "in", xml_ids)] + request.website.website_domain()
-        return View.search(domain).filter_duplicate()
+        self._get_customize_data(self, xml_ids, "ir.ui.view")
+
+    def _get_customize_data(self, keys, model):
+        Data = request.env[model].with_context(active_test=False)
+        if not keys:
+            return Data
+        domain = [("key", "in", keys)] + request.website.website_domain()
+        return Data.search(domain).filter_duplicate()
 
     @http.route(['/website/theme_customize_get'], type='json', auth='user', website=True)
-    def theme_customize_get(self, xml_ids):
-        views = self._get_customize_views(xml_ids)
-        return views.filtered('active').mapped('key')
+    def theme_customize_get(self, keys, model):
```

Comment:
not sure if I have to add a default value for model `model = "ir.ui.view"` as it's a new parameter (for compatibility)

Reply:
@bvr-odoo I confirmed my thought with @rdeodoo about this: no you cannot add arguments to python method even if they are optional, any override would crash or stop working. Instead, here, create an extra route (the old one calling the new one) and add a comment indicating the old one should be removed in master. You have to think "if someone did use this method before in its code, will it still work the same?"

Also: instead of having a "model" argument, make it a boolean to indicate if view or asset, otherwise your method can be used to search on any model for no good reason (still protected via ACL and would also probably simply crash since no filter_duplicate method etc on other models... but never too careful about how a code will evolve.